lib/UR/Object.pm

```diff
@@ -683,6 +685,45 @@ sub is_prunable {
 }
 
 
+sub __rollback__ {
+    my $self = shift;
+
+    my $saved = $self->{db_saved_uncommitted} || $self->{db_committed};
+    unless ($saved) {
+        return UR::Object::delete($self);
+    }
+
+    my $meta = $self->__meta__;
+
+    my $should_rollback = sub {
```

Comment:
I feel like we should extract this to a method.  When `Genome::Disk::Allocation` gets a `__rollback__` it is going to need to duplicate this whole method and then swap `archivable` with `__archivable`.  Maybe an instance method like "give me all the properties that can be saved to the database and have changed".